apps/mull-ui/src/app/pages/profile/edit-profile/edit-profile.tsx

```diff
@@ -1,57 +1,91 @@
 import { MullBackButton } from '../../../components/mull-back-button/mull-back-button';
 import MullButton from '../../../components/mull-button/mull-button';
+import { useToast } from '../../../hooks/useToast';
 import './edit-profile.scss';
 
-const EditProfile = () => {
-  // TODO: Add initial profile image if user has one
-  // TODO: If user does not have an image, set to a default image
-  const [imageURLFile, setImageURLFile] = useState<string>(
-    'https://blog.photofeeler.com/wp-content/uploads/2017/04/are-bumble-profiles-fake-how-many.jpeg'
-  );
+export interface EditProfilePageProps {
+  history: History;
+}
 
-  // Uploaded image file blob
+const EditProfile = ({ history }: EditProfilePageProps) => {
+  const [imageURLFile, setImageURLFile] = useState<string>('');
   const [file, setFile] = useState<File>(null);
-
-  /**
-   * Handles image file uploads
-   * @param {ChangeEvent<HTMLInputElement>} event
-   */
-  const handleFileUpload = (event: ChangeEvent<HTMLInputElement>) => {
-    setImageURLFile(URL.createObjectURL(event.target.files[0]));
-    setFile(event.target.files[0]);
-    formik.setFieldValue('imageFile', event.target.files[0]);
-  };
+  const { data: userData, loading: userLoading } = useUserQuery();
+  const [updateUser] = useUpdateUserMutation();
+  const { updateToast, notifyToast } = useToast();
 
   const formik = useFormik<IProfileEditForm>({
     initialValues: {
       displayName: '',
       description: '',
-      imageFile: null,
+      imageFile: '',
     },
-
     validationSchema: Yup.object({
-      displayName: Yup.string().required('Display name is required'),
-      description: Yup.string().required('Description is required'),
+      displayName: Yup.string()
+        .required('Display name is required')
+        .max(24, 'Display Name must be 24 characters or less.'),
+      description: Yup.string().max(250, 'Description must be 250 characters or less.'),
     }),
-    //TODO: implement form logic
+
     onSubmit: async () => {
       const errors = await formik.validateForm();
       if (isEmpty(errors)) {
-        //TODO: update profile in database
-        console.log('Form successfully submitted');
+        notifyToast('Submitting User Profile...');
+        try {
+          await updateUser({
+            variables: {
+              newAvatar: file,
+              userInput: {
+                id: userData.user.id,
+                name: formik.values.displayName,
+                description: formik.values.description,
+              },
+            },
+          });
+          updateToast(toast.TYPE.SUCCESS, 'Profile Updated');
+          history.push(ROUTES.PROFILE.DISPLAY);
+        } catch (err) {
+          updateToast(toast.TYPE.ERROR, 'Error: Failed To Update Profile');
+          console.error(err);
+        }
       } else {
         formik.setTouched(setNestedObjectValues<FormikTouched<FormikValues>>(errors, true));
       }
```

Comment:
You don't need to call validateForm here and you also don't need the if-else. I had to use it in create event since it was a 2-step process to submit the form. It should validate the form without having to call anything.

Reply:
Done